File: issue-triage/src/jvmTest/java/com/github/karlsabo/devlake/triage/assessment/FakePiProcess.java
package com.github.karlsabo.devlake.triage.assessment;

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Arrays;

public final class FakePiProcess {
    private FakePiProcess() {}

    public static void main(String[] args) throws Exception {
        switch (args[0]) {
            case "assessment" -> assess(args);
            case "sleep" -> sleep(args);
            default -> throw new IllegalArgumentException("Unknown fake pi mode: " + args[0]);
        }
    }

    private static void assess(String[] args) throws Exception {
        Files.write(Path.of(args[1]), Arrays.asList(args).subList(4, args.length));
        Files.copy(System.in, Path.of(args[2]));
        System.out.print(Files.readString(Path.of(args[3])));
    }

    private static void sleep(String[] args) throws Exception {
        Files.writeString(Path.of(args[1]), Long.toString(ProcessHandle.current().pid()));
        System.in.transferTo(java.io.OutputStream.nullOutputStream());
        Thread.sleep(30_000);
    }
}

File: issue-triage/src/jvmTest/kotlin/com/github/karlsabo/devlake/triage/assessment/FakePiProcess.kt
package com.github.karlsabo.devlake.triage.assessment

import java.io.File
import java.io.OutputStream
import java.nio.file.Files
import java.nio.file.Path

internal object FakePiProcess {
    @JvmStatic
    fun main(args: Array<String>) {
        require(args[0] == "sleep") { "Unknown fake pi mode: ${args[0]}" }
        sleep(args)
    }

    private fun sleep(args: Array<String>) {
        Files.writeString(Path.of(args[1]), ProcessHandle.current().pid().toString())
        System.`in`.transferTo(OutputStream.nullOutputStream())
        Thread.sleep(30_000)
    }
}

internal fun fakePiCommand(mode: String, vararg files: Path): List<String> {
    val classpath = listOf(FakePiProcess::class.java, Unit::class.java)
        .map { type -> Path.of(type.protectionDomain.codeSource.location.toURI()).toString() }
        .distinct()
        .joinToString(File.pathSeparator)
    return listOf(
        Path.of(System.getProperty("java.home"), "bin", "java").toString(),
        "-cp",
        classpath,
        FakePiProcess::class.java.name,
        mode,
    ) + files.map(Path::toString)
}
